src/stk/molecular/topology_graphs/topology_graph/optimizer/mchammer.py

```diff
@@ -0,0 +1,218 @@
+
+    Implementation: [1]_
+
+    .. [1] https://github.com/andrewtarzia/MCHammer.
+
+    """
+
+    def __init__(
+        self,
+        step_size=0.25,
+        target_bond_length=1.2,
+        num_steps=500,
+        bond_epsilon=50,
+        nonbond_epsilon=20,
+        nonbond_sigma=1.2,
+        nonbond_mu=3,
+        beta=2,
+        random_seed=None,
+        save_trajectory=False,
+    ):
+        """
+        Initialize an instance of :class:`.MCHammer`.
+
+        Parameters
+        ----------
+        step_size : :class:`float`, optional
+            The relative size of the step to take during step.
+            Defaults to 0.25.
+
+        target_bond_length : :class:`float`, optional
+            Target equilibrium bond length for long bonds to minimize
+            to.
+            Defaults to 1.2 Angstrom.
+
+        num_steps : :class:`int`, optional
+            Number of MC moves to perform.
+            Defaults to 500 steps.
+
+        bond_epsilon : :class:`float`, optional
+            Value of epsilon used in the bond potential in MC moves.
+            Determines strength of the bond potential.
+            Defaults to 50.
+
+        nonbond_epsilon : :class:`float`, optional
+            Value of epsilon used in the nonbond potential in MC moves.
+            Determines strength of the nonbond potential.
+            Defaults to 20.
+
+        nonbond_sigma : :class:`float`, optional
+            Value of sigma used in the nonbond potential in MC moves.
+            Defaults to 1.2.
+
+        nonbond_mu : :class:`float`, optional
+            Value of mu used in the nonbond potential in MC moves.
+            Determines the steepness of the nonbond potential.
+            Defaults to 3.
+
+        beta : :class:`float`, optional
+            Value of beta used in the in MC moves. Beta takes the
+            place of the inverse boltzmann temperature.
+            Defaults to 2.
+
+        random_seed : :class:`int`, optional
+            Random seed to use for MC algorithm. Should only be set
+            if exactly reproducible results are required, otherwise
+            a system-based random seed should be used for proper
+            sampling.
+
+        save_trajectory : :class:`bool`, optional
+            `True` to save MCHammer trajectory and information in
+            :attr:`_trajectory_data`.
+            Defaults to `False`.
+
+        """
+        self._optimizer = mch.Optimizer(
+            step_size=step_size,
+            target_bond_length=target_bond_length,
+            num_steps=num_steps,
+            bond_epsilon=bond_epsilon,
+            nonbond_epsilon=nonbond_epsilon,
+            nonbond_sigma=nonbond_sigma,
+            nonbond_mu=nonbond_mu,
+            beta=beta,
+            random_seed=random_seed,
+        )
+        self._save_trajectory = save_trajectory
+        self._trajectory_data = None
+
+    def optimize(self, state):
+        """
```

Comment:
No need to repeat docstrings of an inhertied method, unless you are changing their content.

Reply:
Done.